src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -1410,11 +1410,17 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
                     left = ASR::down_cast<ASR::expr_t>(ASR::make_ImplicitCast_t(
                         al, left->base.loc, left, ASR::cast_kindType::RealToInteger, dest_type,
                         value));
+                    left = ASR::down_cast<ASR::expr_t>(ASR::make_ImplicitCast_t(
+                        al, left->base.loc, left, ASR::cast_kindType::IntegerToReal, dest_type,
+                        value));
```

Comment:
I'll try to test this soon. It seems it is casting real2int and then int2real. Wouldn't it be better to do something like `float(int(a) // int(b))`?

Reply:
I was trying to do the same initially(in the original code), but since `a` and `b` are float, llvm will use `fdiv` and raise verification error.